nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/AnnotationServiceImpl.java

```diff
@@ -115,15 +93,19 @@ public Annotation getAnnotation(CoreSession s, String annotationId)
                 Map<String, Serializable> props = Collections.singletonMap(CORE_SESSION_PROPERTY, (Serializable) s);
                 PageProvider<DocumentModel> pageProvider = (PageProvider<DocumentModel>) ppService.getPageProvider(
                         GET_ANNOTATIONS_FOR_DOC_PAGEPROVIDER_NAME, null, null, null, props, documentId, xpath);
-                return pageProvider.getCurrentPage().stream().map(Comments::newAnnotation).collect(Collectors.toList());
+
+                return pageProvider.getCurrentPage()
+                                   .stream()
+                                   .map(doc -> (Annotation) toComment(doc))
```

Comment:
You may want to generify `toComment` method to not have this cast here, wdyt?
Something like this should work:
```
public static <C extends Comment> C toComment(DocumentModel)
```